src/openrct2/CmdlineSprite.cpp

```diff
@@ -91,16 +91,16 @@ static bool sprite_file_open(const utf8* path)
     if (spriteFileHeader.num_entries > 0)
     {
         int32_t openEntryTableSize = spriteFileHeader.num_entries * sizeof(rct_g1_element_32bit);
-        rct_g1_element_32bit* openElements = static_cast<rct_g1_element_32bit*>(malloc(openEntryTableSize));
+        std::unique_ptr<rct_g1_element_32bit[]> openElements = std::make_unique<rct_g1_element_32bit[]>(openEntryTableSize);
         if (openElements == nullptr)
         {
             fclose(file);
             return false;
         }
 
-        if (fread(openElements, openEntryTableSize, 1, file) != 1)
+        if (fread(openElements.get(), openEntryTableSize, 1, file) != 1)
         {
-            free(openElements);
+            openElements.reset();
```

Comment:
the joy of unique ptrs is that you don't need to free them. This reset is not required. Its called implicitly when the variable goes out of scope. Same goes for all the other `.reset()`s